### ISSUE TITLE
[web:canvaskit] consider switching to transferToImageBitmap

### ISSUE THREAD

**Issue by yjbanov** [flutter/flutter#140417]
The CanvasKit renderer currently uses `createImageBitmap` [here](https://github.com/flutter/engine/blob/9fda016022a715ee00e1ea1f888b480a871951a0/lib/web_ui/lib/src/engine/canvaskit/surface.dart#L120C61-L120C61) when rendering through an `OffscreenCanvas`. This has two issues:

* Injects an extra layer of asynchrony into frame rendering, likely delaying the frame. The code calling `rasterizeToCanvas` is invoked inside a `requestAnimationFrame`, which is already async by waiting for vsync. The extra `Promise` produced by `createImageBitmap` essentially misses the frame (although not sure if in practice the browser still manages to fit it inside the current frame if there's still budget for it).
* It uses double the amount of bitmap data during window resize. The first bitmap is created and used as the render target for the `OffscreenCanvas`, and a second bitmap is created by calling `createImageBitmap`. Since during resize the bitmap size is volatile, we need to drop both copies and create two new copies to render every frame.

**Comment by knopp** [comment]
Another issue is that `createImageBitmap` seems to be pathologically slow in Firefox. 

<img width="671" alt="Screenshot 2024-01-03 at 22 55 38" src="https://github.com/flutter/flutter/assets/96958/c3553f1c-eb75-471b-8c3f-2f2ee3354b08">


Related: https://github.com/flutter/flutter/issues/140748

**Comment by yjbanov** [comment]
Yes, and unfortunately `transferToImageBitmap` is also [slow in Firefox](https://bugzilla.mozilla.org/show_bug.cgi?id=1788206). Safari is [also slow](https://bugs.webkit.org/show_bug.cgi?id=267291). We stopped using `ImageBitmap` in those browsers for now.
